org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/outline/ModelBuilder.scala

```diff
@@ -57,7 +57,7 @@ object ModelBuilder extends HasLogger {
         } else {
           sb.append("(")
           args.dropRight(1).foreach(x => { renderType(sb, x); sb.append(", ") })
-          sb.setLength(sb.length - 2)
+          sb.setLength(math.max(1, sb.length - 2))
```

Comment:
You could write "(".length and ", ".length instead, that would make it easier to understand.